src/analysis/normed_space/operator_norm.lean

```diff
@@ -1,193 +1,183 @@
+notation E ` →L[`:25 k `] ` F := bounded_linear_map k E F
 
 /-- Coerce bounded linear maps to functions. -/
-instance bounded_linear_maps.to_fun : has_coe_to_fun $ L(E,F) :=
+instance to_fun : has_coe_to_fun $ E →L[k] F :=
 {F :=  λ _, (E → F), coe := (λ f, f.val)}
 
-@[extensionality] theorem ext {A B : L(E,F)} (H : ∀ x, A x = B x) : A = B :=
-set_coe.ext $ funext H
-
-def to_linear_map (A : L(E, F)) : linear_map _ E F :=
-{to_fun := A.val, ..A.property}
+@[extensionality] theorem ext {f g : E →L[k] F} (h : ∀ x, f x = g x) : f = g :=
+set_coe.ext $ funext h
 
-@[simp] lemma bounded_linear_maps.map_zero {A : L(E, F)} : A (0 : E) = 0 :=
-(to_linear_map A).map_zero
+theorem ext_iff {f g : E →L[k] F} : f = g ↔ ∀ x, f x = g x :=
+⟨λ h x, by rintro; rw h, ext⟩
 
-@[simp] lemma bounded_linear_maps.coe_zero {x : E} : (0 : L(E, F)) x = 0 := rfl
+variables (c : k) (f g: E →L[k] F) (h : F →L[k] G) (x u v: E)
 
-@[simp] lemma bounded_linear_maps.smul_coe {A : L(E, F)} {x : E} {c : k} :
-  (c • A) x = c • (A x) := rfl
+def to_linear_map : linear_map _ E F :=
+{to_fun := f.val, ..f.property}
 
-@[simp] lemma bounded_linear_maps.zero_smul {A : L(E, F)} : (0 : k) • A = 0 :=
-by ext; simp
+-- make some straightforward lemmas available to `simp`.
+@[simp] lemma map_zero : f (0 : E) = 0 := (to_linear_map _).map_zero
+@[simp] lemma map_add  : f (u + v) = f u + f v := (to_linear_map _).map_add _ _
+@[simp] lemma map_sub  : f (u - v) = f u - f v := (to_linear_map _).map_sub _ _
+@[simp] lemma map_smul : f (c • u) = c • f u := (to_linear_map _).map_smul _ _
+@[simp] lemma map_neg  : f (-u) = - (f u) := (to_linear_map _).map_neg _
 
-@[simp] lemma bounded_linear_maps.one_smul {A : L(E, F)} : (1 : k) • A = A :=
-by ext; simp
+lemma coe_zero : ((0 : E →L[k] F) : E → F) = 0 := rfl
 
-/-- Bounded linear maps are ... bounded -/
-lemma exists_bound (A : L(E,F)) : ∃ c, 0 < c ∧ ∀ x : E, ∥A x∥ ≤ c * ∥x∥ := A.property.bound
+@[simp] lemma zero_apply : (0 : E →L[k] F) u = 0 := rfl
+@[simp] lemma smul_apply : (c • f) u = c • (f u) := rfl
+@[simp] lemma neg_apply  : (-f) u = - (f u) := rfl
 
-end bounded_linear_maps
+@[simp] lemma zero_smul : (0 : k) • f = 0  := by { ext, simp }
+@[simp] lemma one_smul  : (1 : k) • f = f  := by { ext, simp }
 
-/-
-Now define the operator norm.
+/-- Composition of bounded linear maps. -/
+def comp (g : F →L[k] G) (f : E →L[k] F) : E →L[k] G :=
+⟨_, is_bounded_linear_map.comp g.property f.property⟩
 
-The main task is to show that the operator norm is definite, homogeneous, and satisfies the
-triangle inequality. This is done after a few preliminary lemmas necessary to deal with csupr.
--/
-section operator_norm
+section op_norm
+open lattice set bounded_linear_map
 
-variables [normed_field k] [normed_space k E] [normed_space k F]
-open lattice set
+/-- The operator norm of a bounded linear map is the inf of all its bounds. -/
+def op_norm := real.Inf { c | c ≥ 0 ∧ ∀ x, ∥f x∥ ≤ c * ∥x∥ }
```

Comment:
I'm not up to speed on `real.Inf`. Can't you just write `Inf`, and have type class inference figure out that you mean `real.Inf`?
This might also apply in other cases.

Reply:
That works. Thanks for the catch.

Changes added in a270440f2e98cc5c077cd8daced7ffb3c0129315 .